docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,497 @@
+# Cluster Class
+
+The ClusterClass feature introduces a new way to create clusters which reduces boilerplate required and allows flexible and powerful customization of CAPI-created clusters.
+ClusterClass is implemented on top of pre-existing CAPI interfaces and will 
+
+This tutorial covers using Cluster API and Cluster Class to create a Kubernetes cluster and how to perform a one-touch Kubernetes update procedure using ClusterClass features.
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+This guide covers creating a ClusterClass cluster with the CAPD provider.
+</aside>
+
+## Installation
+
+### Common Prerequisites
+This guide is based on having the same pre-requisites as the CAPI [quick start guide].
+That is:
+- Install and setup [kubectl]  and [clusterctl]
+- Install [Kind] and [Docker]
+
+ClusterClass is currently behind a feature gate that needs to be enabled. To do so run:
+
+```bash
+export  EXP_CLUSTER_RESOURCE_SET=true
+export CLUSTER_TOPOLOGY=true
```

Comment:
I wonder if the env var should actually have been `EXP_CLUSTER_TOPOLOGY` like all our other feature flag env vars.

Reply:
Yeah - I guess it's not all that important, but there's definitely a feature flag drift here - but ClusterClass is also outside the EXP directory so it is implemented in a different way to start.

Reply:
Good point. Okay no idea how it should be :) (@fabriziopandini ?)

Reply:
We should probably not change the feature gate at this point. Would we see instead that new feature gates shouldn't have the EXP prefix instead, given that are all by definition experiments?

Reply:
I agree that the FF prefix isn't really helpful and that we shouldn't change anything now. Usually I think it's a good idea to have some namespacing in env variables (e.g. CAPI_FF prefix), but I think it's too late for that. 3 different formats for this are too much :)